<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Gali Bai" />

<meta name="date" content="2021-06-07" />

<title>scATAC-seq</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MAESTRO 1.5.0</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-gear"></span>
     
    scRNA-seq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="scrna.html">10X PBMC 1k</a>
    </li>
    <li>
      <a href="scrna_stepbystep.html">Custom Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-gear"></span>
     
    scATAC-seq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="scatac.html">10X PBMC 500</a>
    </li>
    <li>
      <a href="scatac_stepbystep.html">Custom Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-gear"></span>
     
    fast scATAC-seq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="fast-scatac.html">multi-sample scATAC-seq</a>
    </li>
    <li>
      <a href="fast-scatac_stepbystep.html">Custom Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a>
    <span class="fas fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/baigal628/MAESTRO_documentation">Source code</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">scATAC-seq</h1>
<h4 class="author">Gali Bai</h4>
<h4 class="date">2021-06-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MAESTRO_documentation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201223code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201223)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201223code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201223)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombaigal628MultiSamplescATACseqtreedf0c2882fe04c6abfaa338b9e6d3ff22f473e62ctargetblankdf0c288a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/baigal628/MultiSample_scATACseq/tree/df0c2882fe04c6abfaa338b9e6d3ff22f473e62c" target="_blank">df0c288</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombaigal628MultiSamplescATACseqtreedf0c2882fe04c6abfaa338b9e6d3ff22f473e62ctargetblankdf0c288a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/baigal628/MultiSample_scATACseq/tree/df0c2882fe04c6abfaa338b9e6d3ff22f473e62c" target="_blank">df0c288</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  MAESTRO_documentation.Rproj
    Untracked:  atac_pbmc_500_nextgem_DiffPeaks.tsv
    Untracked:  atac_pbmc_500_nextgem_cluster.png
    Untracked:  code/Basic_Operations.R
    Untracked:  code/Downstream Analysis.R
    Untracked:  data/fast-scatac/
    Untracked:  data/scatac/
    Untracked:  data/scrna/
    Untracked:  pbmc_1k_v3.lisa/

Unstaged changes:
    Deleted:    MultiSample_scATACseq.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this example, we will be analyzing a scATAC-seq dataset of 500 human peripheral blood mononuclear cells (PBMCs) freely available from 10X Genomics. The raw dataset can be downloaded from the 10X Genomics website. We will show you how to run through the whole MAESTRO pipeline from the raw sequencing fastq files to the final results. We also provide guidance on how to perform custom analysis starting from the pipeline output such as the peak count matrix.</p>
<div id="step-0.-download-the-data-and-prepare-the-environment" class="section level3">
<h3>Step 0. Download the data and prepare the environment</h3>
<p>Please download the raw data from 10X Genomics website.</p>
<pre class="bash"><code>$ wget https://cf.10xgenomics.com/samples/cell-atac/2.0.0/atac_pbmc_500_nextgem/atac_pbmc_500_nextgem_fastqs.tar
$ tar xvf atac_pbmc_500_nextgem_fastqs.tar</code></pre>
<p>Before running MAESTRO, users need to activate the MAESTRO environment.</p>
<pre class="bash"><code>$ conda activate MAESTRO</code></pre>
</div>
<div id="step-1.-configure-the-maestro-workflow" class="section level3">
<h3>Step 1. Configure the MAESTRO workflow</h3>
<p>Initialize the MAESTRO scATAC-seq workflow using <code>MAESTRO scATAC-init</code> command. This will install a Snakefile and a config file in this directory. Here we take the 10X PBMC data as an example. Considering MAESTRO provides built-in immune cell markers, it’s recommended to choose the <code>RP-based</code> cell-type annotation strategy. If the data is not immune-related, users can choose to provide their own marker gene list, or choose to annotate cell types through the <code>peak-based</code> method (see the following argument description for more details), or they can directly skip the annotation step by not setting <code>--annotation</code>.</p>
<pre class="bash"><code>$ MAESTRO scatac-init --platform 10x-genomics --format fastq --species GRCh38 \
--fastq-dir atac_pbmc_500_nextgem_fastqs --fastq-prefix atac_pbmc_500_nextgem \
--cores 16 --directory scATAC/atac_pbmc_500_nextgem --outprefix atac_pbmc_500_nextgem \
--peak-cutoff 100 --count-cutoff 1000 --frip-cutoff 0.2 --cell-cutoff 50 \
--giggleannotation annotations/MAESTRO/giggle.all \
--fasta Refdata_scATAC_MAESTRO_GRCh38_1.1.0/GRCh38_genome.fa \
--whitelist barcodes/737K-cratac-v1.txt \
--rpmodel Enhanced \
--annotation --method RP-based --signature human.immune.CIBERSORT</code></pre>
<p>MAESTRO also allow users to start with bam files with ‘CB’ tag. Here the bam file <code>atac_pbmc_10k_v1_possorted_bam.bam</code> can be downloaded from <a href="https://support.10xgenomics.com/single-cell-atac/datasets/1.1.0/atac_pbmc_10k_v1">10X Genomics</a>.</p>
<pre class="bash"><code>$ MAESTRO scatac-init --platform 10x-genomics --format bam --species GRCh38 \
--bam Data/atac_pbmc_10k_v1_possorted_bam.bam \
--cores 8 --directory 10X_PBMC_10k_MAESTRO_V110 --outprefix 10X_PBMC_10k \
--peak-cutoff 100 --count-cutoff 1000 --frip-cutoff 0.2 --cell-cutoff 50 \
--rpmodel Enhanced \
--giggleannotation annotations/MAESTRO/giggle.all \
--annotation --method both --signature human.immune.CIBERSORT --clusterpeak</code></pre>
<p>Fragment file generated by Cellranger ATAC is also supported as the input of the workflow. Here the fragment file <code>atac_pbmc_10k_v1_fragments.tsv.gz</code> can be downloaded from <a href="https://support.10xgenomics.com/single-cell-atac/datasets/1.1.0/atac_pbmc_10k_v1">10X Genomics</a>.</p>
<pre class="bash"><code>$ MAESTRO scatac-init --platform 10x-genomics --format fragments --species GRCh38 \
--frag Data/atac_pbmc_10k_v1_fragments.tsv.gz \
--cores 8 --directory 10X_PBMC_10k_MAESTRO_V110 --outprefix 10X_PBMC_10k \
--peak-cutoff 100 --count-cutoff 1000 --frip-cutoff 0.2 --cell-cutoff 50 \
--rpmodel Enhanced \
--giggleannotation annotations/MAESTRO/giggle.all \
--annotation --method both --signature human.immune.CIBERSORT --clusterpeak</code></pre>
<p>To get a full description of command-line options, please use the command</p>
<pre class="bash"><code>
$ MAESTRO scatac-init -h
</code></pre>
<div id="here-we-list-all-the-arguments-and-their-description." class="section level4">
<h4>Here we list all the arguments and their description.</h4>
<p><strong>Input files arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--platform</code></td>
<td>{10x-genomics,sci-ATAC-seq,microfluidic} Platform of single cell ATAC-seq. DEFAULT: 10x-genomics.</td>
</tr>
<tr class="even">
<td><code>--format</code></td>
<td>{fastq,bam,fragments} The format of input files. Users can start with sequencing fastq files, bam files with CB tag or fragments.tsv.gz file generated by CellRanger ATAC. If the platform is 10x-genomics, users can start with sequencing fastq files, bam files with CB tag or fragments.tsv.gz file generated by CellRanger ATAC. If the platform is sci-ATAC-seq, users can start with sequencing fastq files, bam files with CB tag. If the platform is microfluidic, users can only start with sequencing fastq files. If the format is ‘fastq’, –fastq-dir and –fastq-prefix need to be set. If the format is ‘bam’, –bam needs to be set. If the format is ‘fragments’, –frag needs to be set. DEFAULT: fastq.</td>
</tr>
<tr class="odd">
<td><code>--fastq-dir</code></td>
<td>Directory where fastq files are stored.</td>
</tr>
<tr class="even">
<td><code>--fastq-prefix</code></td>
<td>Sample name of fastq file (required for the platform of ‘10x-genomics’ or ‘sci-ATAC-seq’). When the platform is ‘10x-genomics’, if there is a file named pbmc_1k_v2_S1_L001_I1_001.fastq.gz, the prefix is ‘pbmc_1k_v2’. If the platform is ‘sci-ATAC-seq’, there are two ways to provide fastq files. The first is to provide pair-end sequencing results that contain two fastq files – prefix_1.fastq and prefix_2.fastq. If in this way, the barcode for each read needs to be included in the reads ID (the first line of each read) in the format of ‘<span class="citation">@ReadName:CellBarcode:OtherInformation</span>’. For example, <span class="citation">@rd.1:TCTCCCGCCGAGGCTGACTGCATAAGGCGAAT:SHEN-MISEQ02:1:1101:15311:1341</span>. The other way is to provide 10x-like fastq files which should contain three fastq files – prefix_R1.fastq, prefix_R2.fastq and prefix_R3.fastq. In this way, read1, barcode and read2 are associated with R1, R2, R3, respectively.</td>
</tr>
<tr class="odd">
<td><code>--bam BAM</code></td>
<td>The bam file with cell barcode (CB) tag (required when the format is set as ‘bam’).</td>
</tr>
<tr class="even">
<td><code>--frag FRAG</code></td>
<td>The fragment file generated by CellRanger ATAC (required when the format is set as ‘fragments’). For example, fragments.tsv.gz.</td>
</tr>
<tr class="odd">
<td><code>--species</code></td>
<td>{GRCh38,GRCm38} Specify the genome assembly (GRCh38 for human and GRCm38 for mouse). DEFAULT: GRCh38.</td>
</tr>
</tbody>
</table>
<p><strong>Running and output arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--cores</code></td>
<td>The number of cores to use. DEFAULT: 8.</td>
</tr>
<tr class="even">
<td><code>--directory</code></td>
<td>Path to the directory where the workflow shall be initialized and results shall be stored. DEFAULT: MAESTRO.</td>
</tr>
<tr class="odd">
<td><code>--outprefix</code></td>
<td>Prefix of output files. DEFAULT: MAESTRO.</td>
</tr>
</tbody>
</table>
<p><strong>Quality control arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--peak-cutoff</code></td>
<td>Minimum number of peaks included in each cell. DEFAULT: 100.</td>
</tr>
<tr class="even">
<td><code>--count-cutoff</code></td>
<td>Cutoff for the number of count in each cell. DEFAULT: 1000.</td>
</tr>
<tr class="odd">
<td><code>--frip-cutoff</code></td>
<td>Cutoff for fraction of reads in promoter in each cell. DEFAULT: 0.2.</td>
</tr>
<tr class="even">
<td><code>--cell-cutoff</code></td>
<td>Minimum number of cells covered by each peak. DEFAULT: 10.</td>
</tr>
</tbody>
</table>
<p><strong>Reference genome arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--giggleannotation</code></td>
<td>Path of the giggle annotation file required for regulator identification. Please download the annotation file from <a href="http://cistrome.org/~galib/giggle.all.tar.gz">here</a> and decompress it. <strong>Note:</strong> If the version of MAESTRO is higher than v1.2.0, please update the giggle index to the <a href="http://cistrome.org/~galib/giggle.all.tar.gz">latest</a>.</td>
</tr>
<tr class="even">
<td><code>--fasta</code></td>
<td>Genome fasta file for minimap2. Users can just download the fasta file for <a href="http://cistrome.org/~galib/Refdata_scATAC_MAESTRO_GRCh38_1.1.0.tar.gz">human</a> and <a href="http://cistrome.org/~galib/Refdata_scATAC_MAESTRO_GRCm38_1.1.0.tar.gz">mouse</a> from CistromDB and decompress them. For example, <code>--fasta Refdata_scATAC_MAESTR O_GRCh38_1.1.0/GRCh38_genome.fa</code>.</td>
</tr>
</tbody>
</table>
<p><strong>Barcode library arguments, only for the platform of ‘sci-ATAC-seq’:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--whitelist</code></td>
<td>If the platform is ‘sci-ATAC-seq’ or ‘10x-genomics’, please specify the barcode library (whitelist) so that the pipeline can correct cell barcodes with 1 base mismatched. Otherwise, the pipeline will automatically output the barcodes with enough read count (&gt;1000). The 10X Chromium whitelist file can be found inside the CellRanger-ATAC distribution. For example, in CellRanger-ATAC 1.1.0, the whitelist is ‘cellranger-atac-1.1.0/cellranger-atac-cs/1.1.0/lib/python/barcodes/737K-cratac-v1.txt’.</td>
</tr>
</tbody>
</table>
<p><strong>Customized peak arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--custompeak</code></td>
<td>Whether or not to provide custom peaks. If set, users need to provide the file location of peak file through <code>--custompeak-file</code> and then MAESTRO will merge the custom peak file and the peak file called from all fragments using MACS2. By default (not set), the pipeline will use the peaks called using MACS2.</td>
</tr>
<tr class="even">
<td><code>--custompeak-file</code></td>
<td>If <code>--custompeak</code> is set, please provide the file location of custom peak file. The peak file is BED formatted with tab-separated. The first column is the chromosome, the second is chromStart, and the third is chromEnd.</td>
</tr>
<tr class="odd">
<td><code>--shortpeak</code></td>
<td>Whether or not to call peaks from short fragments (shorter than 150bp). If set, MAESTRO will merge the peaks called from all fragments and those called from short fragments, and then use the merged peak file for further analysis. If not (by default), the pipeline will only use peaks called from all fragments.</td>
</tr>
<tr class="even">
<td><code>--clusterpeak</code></td>
<td>Whether or not to call peaks by cluster. If set, MAESTRO will split the bam file according to the clustering result, and then call peaks for each cluster. By default (not set), MAESTRO will skip this step.</td>
</tr>
</tbody>
</table>
<p><strong>Gene score arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--rpmodel</code></td>
<td>{Simple,Enhanced} The RP model to use to calaculate gene score. For each gene, simple model sums over the impact of all regulatory elements within the up/dowm-stream of TSS. On the basis of simple model, enhanced model gives the regulatory elements within the exon region a higher weight, and also excludes the regulatory elements overlapped with another gene (the promoter and exon of a nearby gene). See the MAESTRO paper for more details. DEFAULT: Enhanced.</td>
</tr>
<tr class="even">
<td><code>--genedistance</code></td>
<td>Gene score decay distance, could be optional from 1kb (promoter-based regulation) to 10kb (enhancer-based regulation). DEFAULT: 10000.</td>
</tr>
</tbody>
</table>
<p><strong>Cell-type annotation arguments:</strong></p>
<table>
<colgroup>
<col width="45%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>--annotation</code></td>
<td>Whether or not to perform ccell-type annotation. By default (not set), MAESTRO will skip the step of cell-type annotation. If set, please specify the method of cell-type annotation through <code>--method</code>.</td>
</tr>
<tr class="even">
<td><code>--method</code></td>
<td>{RP-based,peak-based,both} Method to annotate cell types. MAESTRO provides two strategies to annotate cell types for scATAC-seq data. Users can choose from ‘RP-based’ and ‘peak-based’, or choose to run both of them. One is based on gene regulatory potential predicted by RP model. Another is based on the bulk chromatin accessibility data from Cistrome database. If ‘RP-based’ is set, MAESTRO performs the cell-type annotation using the gene regulatory potential to represent gene expression, and the logFC of gene regulatory potential between one cluster and all the other cells is used to calculate the gene signature scores. If ‘peak-based’ is set, MAESTRO utilizes GIGGLE to evaluate the enrichment of bulk chromatin accessibility peaks on cluster-specific peaks from scATAC-seq data, and then transfers the Cistrome cluster identity from the most enriched bulk chromatin accessibility data as the cell-type annotation for the scATAC-seq cluster. See the MAESTRO paper for more details. DEFAULT: RP-based.</td>
</tr>
<tr class="odd">
<td><code>--signature</code></td>
<td>Cell signature file used to annotate cell types (required when method is set as ‘RP-based’). MAESTRO provides several sets of built-in cell signatures. Users can choose from [‘human.immune.CIBERSORT’, ‘mouse.brain.ALLEN’, ‘mouse.all.facs.TabulaMuris’, ‘mouse.all.droplet.TabulaMuris’]. Custom cell signatures are also supported. In this situation, users need to provide the file location of cell signatures, and the signature file is tab-seperated without header. The first column is cell type, and the second column is signature gene. DEFAULT: human.immune.CIBERSORT.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="step-2.-run-maestro" class="section level3">
<h3>Step 2. Run MAESTRO</h3>
<p>Before running the workflow, please check the <code>config.yaml</code> and see if it is configured correctly. Once configured, users can use snakemake to run the workflow.</p>
<pre class="bash"><code>$ cd atac_pbmc_500_nextgem
$ snakemake -np
$ nohup snakemake --cores 12 &gt; atac_pbmc_500_nextgem.out &amp;</code></pre>
</div>
<div id="step-3.-understand-the-final-output-files" class="section level3">
<h3>Step 3. Understand the final output files</h3>
<p>The whole pipeline in this example takes about 6 hours with 10 cores. Here, we assume users have run MAESTRO from fastq files successfully. An output directory is specified in the run call, and will contain several useful outputs as described below.</p>
<pre class="bash"><code>$ ls Result/
atac_pbmc_500_nextgem_scATAC_report.html  Analysis  Benchmark  Log  minimap2  QC</code></pre>
<div id="output-files" class="section level4">
<h4>Output files</h4>
<ul>
<li><strong>minimap2:</strong> The <code>minimap2</code> directory contains all the mapping results generated by <code>minimap2</code>. <code>fragments_corrected_count.tsv</code> is a BED-like tabular file, in which each line represents a unique ATAC-seq fragment captured by the assay. The 5 columns in the file are chrom, chromStart, chromEnd, barcode, and duplicateCount, respectively. All the barcodes have been corrected according to the whitelist users provide. <code>fragments_corrected_count.tsv</code> is used for counting peaks in the downstream analysis.</li>
<li><strong>QC:</strong> The <code>QC</code> directory contains quality control analysis results of scATAC-seq data, including the barcode filtering table <code>outprefix_scATAC_validcells.txt</code>, filtered count matrix <code>outprefix_filtered_peak_count.h5</code> and other QC results. The filtered count matrix is generated according to <code>singlecell.txt</code> and the parameters like <code>--count-cutoff</code> and <code>--frip-cutoff</code> that users provide. <code>singlecell.txt</code> is a tabular file, which provides QC information associated with the fragments per barcode. The file contains three columns, representing barcode, number of fragments and number of fragments overlapping promoter regions.</li>
<li><strong>Analysis:</strong> The <code>Analysis</code> directory contains MACS2 peak calling result, peak count table and gene score table, as well as clustering result, cell type annotation result and driver transcription factor identification result, which we will introduce in the step-by-step analysis.</li>
<li><strong>Benchmark:</strong> The <code>Benchmark</code> directory stores benchmark files for all rules in Snakefile, each of which contains a tab-separated table of run times and memory usage in MiB.</li>
<li><strong>Log:</strong> The Log directory contains the log files generated in the pipeline analysis.</li>
<li><strong>outprefix_scATAC_report.html:</strong> The <code>outprefix_scATAC_report.html</code> file summarizes all the results in an HTML based document. The summary HTML for the 10X_PBMC_10k example can be found <a href="http://cistrome.org/~galib/10X_PBMC_10k_scATAC_report.html">here</a>.</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        knitr_1.33        magrittr_2.0.1    R6_2.5.0         
 [5] rlang_0.4.10      fansi_0.4.2       stringr_1.4.0     tools_4.0.5      
 [9] xfun_0.22         utf8_1.2.1        git2r_0.28.0      jquerylib_0.1.4  
[13] htmltools_0.5.1.1 ellipsis_0.3.1    rprojroot_2.0.2   yaml_2.2.1       
[17] digest_0.6.27     tibble_3.1.1      lifecycle_1.0.0   crayon_1.4.1     
[21] later_1.2.0       sass_0.3.1        vctrs_0.3.7       promises_1.2.0.1 
[25] fs_1.5.0          glue_1.4.2        evaluate_0.14     rmarkdown_2.7    
[29] stringi_1.5.3     bslib_0.2.4       compiler_4.0.5    pillar_1.6.0     
[33] jsonlite_1.7.2    httpuv_1.6.0      pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
