---
title: "Downstream Analysis"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

### About

Custom analysis can be conducted using either MAESTRO or Seurat package. In this study, benchmarking experiments were designed to systematically test the effect of varying processing times on single-cell transcriptome and **epigenome profiles** from healthy and diseased donors. Peripheral blood mononuclear cells (PBMC) were isolated from healthy donors and from patients affected with chronic lymphocytic leukemia (CLL). Samples were either preserved immediately (0 h) or after 8 and 24 h.

**Goal**: Evaluating the effect of sampling time on scATAC-seq chromatin accessibility profiles.

### Load Required Libraries
```{r, required-packages}
library(Seurat)
library(tidyr)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(presto)
```

### Read Seurta object
```{r, read-rds}
all_samples_scATAC_Object <- readRDS("./data/all_samples_scATAC_Object.rds")
ATAC <- all_samples_scATAC_Object$ATAC
ATAC
DefaultAssay(ATAC) <- 'ATAC'
ATAC[['ATAC']]
```


### Extract Sample Information
```{r, extract-sample-information}
df <- as.data.frame(do.call(rbind, strsplit(row.names(ATAC@meta.data), '@')))
ATAC@meta.data[,c("sample", "hours", "cell")] <- df  %>% separate(V1,into = c("sample", "hours"), 
                                                                  sep = "(?<=[A-Za-z])(?=[0-9])")
head(ATAC@meta.data)
```


### Plot Umap on Default Parameters (Res=0.6)

```{r, UMAP-plot_default}
p1 <-DimPlot(ATAC, label = TRUE, reduction = "umap", group.by = "seurat_clusters", repel=T, pt.size = 0.5, label.size = 3) + labs(title = "Clustering of PBMCs from all samples")

p2 <- DimPlot(ATAC, label = TRUE, reduction = "umap", group.by = "assign.ident", repel=T, pt.size = 0.5, label.size = 3) + labs(title = "Annotaion of PBMCs from all samples")

p3 <- DimPlot(ATAC, label = FALSE, reduction = "umap", group.by = "hours", repel=T, pt.size = 0.5, label.size = 2.5, cols = brewer.pal(3,"Set2")) + labs(title = "Clustering of PBMCs in different time points")

p4 <- DimPlot(ATAC, label = FALSE, reduction = "umap", group.by = "sample", repel=T, pt.size = 0.5, label.size = 2.5, cols = brewer.pal(3,"Pastel2")) + labs(title = "Clustering of PBMCs for treatment and control")
p1
p2
p3
p4
```

**Conclusion:** From the above figures, we can easily observe similar results as described in the paper. A remarkable effect of the sampling time on single-cell epigenome profiles are lying across all blood cell subtypes from healthy donors and CLL patient samples.

### Find Differencially Accessible Peaks

**Between Clusters**
```{r, da-peaks-1}
da_peaks_C <- wilcoxauc(ATAC, 'seurat_clusters', seurat_assay = 'ATAC', assay = 'data')
##Use the 'data slots' under 'ATAC' peak assay
head(da_peaks_C)

```

**Between Samples**
```{r, da-peaks-2}
da_peaks_S <- wilcoxauc(ATAC, 'sample', seurat_assay = 'ATAC', assay = 'data')
head(da_peaks_S)
```

**Between Sampling Time**
```{r, da-peaks-3}
da_peaks_S <- wilcoxauc(ATAC, 'hours', seurat_assay = 'ATAC', assay = 'data')
head(da_peaks_S)
```

There are many other ways you can customize the analysis based on your research purpose. Also, [MAESTRO R Package](https://github.com/liulab-dfci/MAESTRO/blob/master/example/ATAC_infrastructure_10x/ATAC_infrastructure_10x.md) have multiple functions you may also want to have a try. Single-sample tutorials can be applied similarly on a multi-sample object.
